src/Polynomials/MonomialBases.jl

```diff
@@ -146,48 +148,115 @@ function evaluate!(cache,f::MonomialBasis{D,T},x::AbstractVector{<:Point}) where
   r.array
 end
 
-function return_cache(
+function _return_cache(
   fg::FieldGradientArray{1,MonomialBasis{D,V}},
-  x::AbstractVector{<:Point}) where {D,V}
+  x::AbstractVector{<:Point},
+  ::Type{T},
+  TisbitsType::Val{true}) where {D,V,T}
 
   f = fg.fa
   @assert D == length(eltype(x)) "Incorrect number of point components"
   np = length(x)
   ndof = length(f.terms)*num_components(V)
-  xi = testitem(x)
-  T = gradient_type(V,xi)
   n = 1 + _maximum(f.orders)
   r = CachedArray(zeros(T,(np,ndof)))
   v = CachedArray(zeros(T,(ndof,)))
   c = CachedArray(zeros(eltype(T),(D,n)))
   g = CachedArray(zeros(eltype(T),(D,n)))
-  (r, v, c, g)
+  (r,v,c,g)
 end
 
-function evaluate!(
+function _return_cache(
```

Comment:
Reduce code replication here. Define this function as (be careful here, I did not execute it, it may contain errors, but just for sharing the idea with you):

```julia
function _return_cache(
  fg::FieldGradientArray{1,MonomialBasis{D,V}},
  x::AbstractVector{<:Point},
  ::Type{T},
  TisbitsType::Val{false}) where {D,V,T}

  cache = _return_cache(fg,x,T,Val{true}())
  z = CachedArray(zeros(eltype(T),D))
  (cache...,z)
end
```

Reply:
done

Reply:
Thanks!